src/main/java/org/traccar/api/resource/PermissionsResource.java

```diff
@@ -57,27 +59,48 @@ private void checkPermission(Permission permission, boolean link) {
 
     @POST
     public Response add(LinkedHashMap<String, Long> entity) throws SQLException, ClassNotFoundException {
+        return add(Collections.singletonList(entity));
+    }
+
+    @Path("bulk")
+    @POST
+    public Response add(List<LinkedHashMap<String, Long>> entities) throws SQLException, ClassNotFoundException {
         Context.getPermissionsManager().checkReadonly(getUserId());
-        Permission permission = new Permission(entity);
-        checkPermission(permission, true);
-        Context.getDataManager().linkObject(permission.getOwnerClass(), permission.getOwnerId(),
-                permission.getPropertyClass(), permission.getPropertyId(), true);
-        LogAction.link(getUserId(), permission.getOwnerClass(), permission.getOwnerId(),
-                permission.getPropertyClass(), permission.getPropertyId());
-        Context.getPermissionsManager().refreshPermissions(permission);
+        for (LinkedHashMap<String, Long> entity: entities) {
+            Permission permission = new Permission(entity);
+            checkPermission(permission, true);
+            Context.getDataManager().linkObject(permission.getOwnerClass(), permission.getOwnerId(),
+                    permission.getPropertyClass(), permission.getPropertyId(), true);
+            LogAction.link(getUserId(), permission.getOwnerClass(), permission.getOwnerId(),
+                    permission.getPropertyClass(), permission.getPropertyId());
+        }
+        // we assume all permissions are of same type so we use the first one for refreshing
```

Comment:
I think we should probably throw an exception if permissions are not the same type.